### ISSUE TITLE
Sorting problem in parent view (sorting mode 4)

### ISSUE THREAD

**Issue by qzminski** [contao/core#8292]
Tested on Contao 3.5.9.

#### Problem 1

There is a sorting problem in the parent view (sorting mode 4) when you get into that view via ```Save and edit``` button. It works normally when you enter the page via edit button (yellow pencil icon) in the parent view.

The issue is reproducable in the official Contao demo:

1. Go to the Events, edit the calendar events.
2. Create a new event and click the ```Save and edit``` button.
3. Create two new content elements (type does not matter).
4. Try to change the sorting of content elements - will not work.

#### Problem 2

On our system there was also a problem with automatic change of pid of content element with ID of the current parent ID. Example:

We have created a calendar event with ID 30 and clicked the ```Save and edit``` button. Then we created two content elements. After changing the sorting via drag'n'drop and after refreshing the page, it occurred  that Contao behind scenes found ```tl_content.id=30``` and set its ```tl_content.pid``` to 30 (ID of calendar event)!

I am pretty sure this would happen on Contao official demo as well but I can't reproduce it now, probably because I can't pair the newly created event ID with the content element that has the same ID.

#### Hints

I am almost sure that it is related to the URL parameters. If we compare the URL of in theory the same page, in fact we get two different param sets:

URL achieved by clicking the ```Save and edit``` button:
```
contao/main.php?do=calendar&table=tl_content&id=9&mode=2&pid=3
```

URL achieved by clicking the edit button in parent view:
```
contao/main.php?do=calendar&table=tl_content&id=9
```

They are clearly not the same and the major difference is the ```pid``` parameter in the first one. That could be the cause of all evil.

I also think the problem may lay somewhere here, at least at some point because it takes the current request (which has that potentially wrong ```pid``` parameter):

https://github.com/contao/core/blob/master/assets/contao/js/core-uncompressed.js#L1376-L1393

**Comment by Toflar** [comment]
I can confirm the issue. This is a nasty one! :(

**Comment by leofeyer** [comment]
The following change should fix the issue:

```diff
diff --git a/system/modules/core/drivers/DC_Table.php b/system/modules/core/drivers/DC_Table.php
index 5e1e3c6..d38c1b5 100644
--- a/system/modules/core/drivers/DC_Table.php
+++ b/system/modules/core/drivers/DC_Table.php
@@ -2113,7 +2113,7 @@ class DC_Table extends \DataContainer implements \listable, \editable
 				\System::setCookie('BE_PAGE_OFFSET', 0, 0);
 
 				$strUrl = $this->addToUrl($GLOBALS['TL_DCA'][$this->strTable]['list']['operations']['edit']['href'], false);
-				$strUrl = preg_replace('/(&amp;)?(s2e|act)=[^&]*/i', '', $strUrl);
+				$strUrl = preg_replace('/(&amp;)?(s2e|act|mode|pid)=[^&]*/i', '', $strUrl);
 
 				$this->redirect($strUrl);
 			}
```

@qzminski Can you please confirm?

**Comment by Toflar** [comment]
I can confirm the bugfix fixes both, problems 1 and 2!

**Comment by leofeyer** [comment]
Fixed in ef87e7b9d5a9f31fd019fe65b18430cd2d4742f0.
